src/decoder.rs

```diff
@@ -545,117 +544,145 @@ pub enum ScopeEntry {
     Decoder(Decoder),
 }
 
-pub struct Scope<'a> {
-    parent: Option<&'a Scope<'a>>,
-    names: Vec<Label>,
-    entries: Vec<ScopeEntry>,
+pub enum Scope<'a> {
```

Comment:
It might be more ergonomic if we had

```rust
impl<'a> From<&'a MultiScope<'a>> for Scope<'a> {}
impl<'a> From<SingleScope<'a>> for Scope<'a> {}
impl<'a> From<DecoderScope<'a>> for Scope<'a> {}
```

which would allow us to unify on an `.into()` pattern for constructing the wrapper instead of explicitly naming the variant in question. Not sure how useful that might be, but it could make things easier, perhaps.

Reply:
I think I'll hold off on this for now as scope might require some other changes involving traits soon.